tests/shared/drive-constraints.spec.js

```diff
@@ -556,4 +557,240 @@ describe('Shared: DriveConstraints', function() {
+          size: 1000000000,
+          recommendedDriveSize: 2000000000
+        });
+
+        m.chai.expect(result).to.deep.equal([]);
+      });
+
+    });
+
+    describe('given a locked drive and image is null', () => {
+
+      it('should return locked drive error', function() {
+        this.drive.protected = true;
+
+        const result = constraints.getDriveImageCompatibilityStatuses(this.drive, null);
+        const expectedTuples = [ [ 'ERROR', 'LOCKED' ] ];
+
+        expectStatusTypesAndMessagesToBe(result, expectedTuples);
+      });
+
+    });
+
+    describe('given a system drive and image is null', () => {
+
+      it('should return system drive warning', function() {
+        this.drive.system = true;
+
+        const result = constraints.getDriveImageCompatibilityStatuses(this.drive, null);
+        const expectedTuples = [ [ 'WARNING', 'SYSTEM' ] ];
+
+        expectStatusTypesAndMessagesToBe(result, expectedTuples);
+      });
+
+    });
+
+    describe('given the drive contains the image and the drive is locked', () => {
+
+      it('should return the contains-image drive error by precedence', function() {
+        const result = constraints.getDriveImageCompatibilityStatuses(this.drive, {
+          path: `${this.mountpoint}${this.separator}rpi.img`,
+          size: 5000000000
+        });
+        const expectedTuples = [ [ 'ERROR', 'CONTAINS_IMAGE' ] ];
+
+        expectStatusTypesAndMessagesToBe(result, expectedTuples);
+      });
+
+    });
+
+    describe('given a locked and too small drive', () => {
+
+      it('should return the locked error by precedence', function() {
+        this.drive.protected = true;
+
+        const result = constraints.getDriveImageCompatibilityStatuses(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 5000000000
+        });
+        const expectedTuples = [ [ 'ERROR', 'LOCKED' ] ];
+
+        expectStatusTypesAndMessagesToBe(result, expectedTuples);
+      });
+
+    });
+
+    describe('given a too small and system drive', () => {
+
+      it('should return the too small drive error by precedence', function() {
+        this.drive.system = true;
+
+        const result = constraints.getDriveImageCompatibilityStatuses(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 5000000000
+        });
+        const expectedTuples = [ [ 'ERROR', 'TOO_SMALL' ] ];
+
+        expectStatusTypesAndMessagesToBe(result, expectedTuples);
+      });
+
+    });
+
+    describe('given a system drive and not recommended drive size', () => {
+
+      it('should return both warnings', function() {
+        this.drive.system = true;
+
+        const result = constraints.getDriveImageCompatibilityStatuses(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000,
+          recommendedDriveSize: 5000000000
+        });
+        const expectedTuples = [ [ 'WARNING', 'SYSTEM' ], [ 'WARNING', 'SIZE_NOT_RECOMMENDED' ] ];
```

Comment:
Given that this is checking against a list of tuples (rather than a set of tuples) does this mean this test would fail if `.getDriveImageCompatibilityStatuses` returned a list with 'SIZE_NOT_RECOMMENDED' before 'SYSTEM'? IMHO it might be nice if the test would still pass in that case?